### PR TITLE
[CoreBundle] Unification of services names

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 1.14
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | #17093
| License         | MIT

### CODE REVIEW

File: UPGRADE-1.14.md [hunk 1]
```diff
@@ -35,337 +35,563 @@
...
+    | `sylius.channel_collector`                                                                                          | `sylius.collector.channel`                                                                          |
+    | **CoreBundle**                                                                                                      |                                                                                                     |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Calculator\FixedDiscountPriceCalculator`                                 | `sylius.calculator.catalog_promotion.fixed_discount`                                                |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Calculator\PercentageDiscountPriceCalculator`                            | `sylius.calculator.catalog_promotion.percentage_discount_price`                                     |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\CatalogPromotionEligibilityChecker`                              | `sylius.eligibility_checker.catalog_promotion`                                                      |
```
Comment (GSadee) [inline UPGRADE-1.14.md:197]:
```suggestion
    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\CatalogPromotionEligibilityChecker`                              | `sylius.checker.catalog_promotion_eligibility`                                                      |
```
but in the separate PR 🖖🏻

File: UPGRADE-1.14.md [hunk 2]
```diff
@@ -35,337 +35,563 @@
...
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\InForProductScopeVariantChecker`                                 | `sylius.checker.catalog_promotion.in_for_product_scope_variant`                                     |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\InForTaxonsScopeVariantChecker`                                  | `sylius.checker.catalog_promotion.in_for_taxons_scope_variant`                                      |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\InForVariantsScopeVariantChecker`                                | `sylius.checker.catalog_promotion.in_for_variants_scope_variant`                                    |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\ApplyCatalogPromotionsOnVariantsHandler`                  | `sylius.command_handler.catalog_promotion.apply_variants`                                           |
+    | `Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\DisableCatalogPromotionHandler`                           | `sylius.command_handler.catalog_promotion.disabled`                                                 |
```
Comment (GSadee) [inline UPGRADE-1.14.md:202]:
```suggestion
    | `Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\DisableCatalogPromotionHandler`                           | `sylius.command_handler.catalog_promotion.disable`                                                 |
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -73,34 +84,40 @@
...
             <tag name="sylius.payment_method_resolver" type="channel_based" label="sylius.payment_methods_resolver.channel_based" priority="1" />
         </service>
+        <service id="sylius.resolver.payment_methods.channel_based" alias="sylius.payment_methods_resolver.channel_based" />
 
         <service id="sylius.payment_method_resolver.default" class="Sylius\Component\Core\Resolver\DefaultPaymentMethodResolver">
             <argument type="service" id="sylius.repository.payment_method" />
         </service>
-        <service id="Sylius\Component\Payment\Resolver\DefaultPaymentMethodResolverInterface" alias="sylius.payment_method_resolver.default" />
+        <service id="sylius.resolver.payment_method.default" alias="sylius.payment_method_resolver.default" />
+<!--        <service id="Sylius\Component\Payment\Resolver\DefaultPaymentMethodResolverInterface" alias="sylius.payment_method_resolver.default" />-->
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml:107]: Why?

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/catalog_promotion/command_handlers.xml [hunk 4]
```diff
@@ -11,40 +11,49 @@
...
         </service>
+        <service id="sylius.command_handler.catalog_promotion.apply_variants" alias="Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\ApplyCatalogPromotionsOnVariantsHandler" />
 
         <service id="Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\DisableCatalogPromotionHandler">
             <argument type="service" id="sylius.repository.catalog_promotion" />
-            <argument type="service" id="Sylius\Bundle\CoreBundle\CatalogPromotion\Processor\AllProductVariantsCatalogPromotionsProcessorInterface"/>
+            <argument type="service" id="sylius.processor.catalog_promotion.all_product_variant"/>
             <tag name="messenger.message_handler" bus="sylius.command_bus" />
         </service>
+        <service id="sylius.command_handler.catalog_promotion.disabled" alias="Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\DisableCatalogPromotionHandler" />
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/catalog_promotion/command_handlers.xml:36]:
```suggestion
        <service id="sylius.command_handler.catalog_promotion.disable" alias="Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler\DisableCatalogPromotionHandler" />
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/catalog_promotion/checkers.xml [hunk 5]
```diff
@@ -21,23 +25,29 @@
         >
             <argument type="tagged_locator" tag="sylius.catalog_promotion.variant_checker" index-by="type" />
         </service>
+        <service id="sylius.checker.catalog_promotion.product_variant_for_catalog_promotion_eligibility"
+                 alias="Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\ProductVariantForCatalogPromotionEligibilityInterface" />
 
         <service id="Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\CatalogPromotionEligibilityChecker">
             <argument type="tagged_iterator" tag="sylius.catalog_promotion.criteria" />
         </service>
+        <service id="sylius.eligibility_checker.catalog_promotion" alias="Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\CatalogPromotionEligibilityChecker" />
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/catalog_promotion/checkers.xml:34]:
```suggestion
        <service id="sylius.checker.catalog_promotion_eligibility" alias="Sylius\Bundle\CoreBundle\CatalogPromotion\Checker\CatalogPromotionEligibilityChecker" />
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml [hunk 6]
```diff
@@ -11,53 +11,57 @@
...
-</services>
+<container
+    xmlns="http://symfony.com/schema/dic/services"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <defaults public="true" />
+
+        <service id="sylius.listener.user_mailer_listener" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener" public="false">
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml:22]:
```suggestion
        <service id="sylius.listener.user_mailer" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener" public="false">
```
or
```suggestion
        <service id="sylius.listener.mailer.user" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener" public="false">
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Karol! :1st_place_medal:

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml (hunk 3 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/CoreBundle/Resources/config/services.xml:107]: Mistake

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml (hunk 6 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml:22]: This service definition overrides the service from `UserBundle`.

**Pull request merged**